src/components/accounts/SetupImplicit.js

```diff
@@ -0,0 +1,289 @@
+            const confirm = confirm('WARNING! If you have funded your account, press "Cancel".\n\nIf you would like to start over with a new account, press "OK".')
+            if (confirm) {
+                localStorage.removeItem(IMPLICIT_ACCOUNT_KEY)
+                this.refreshData()
+                this.props.history.push('/create-implicit')
+            } else {
+                this.props.history.push('/create-implicit/fund')
+            }
+        }
+        if (this.props.location.pathname === '/create-implicit/fund' && state.snackBarMessage !== 'setupSeedPhrase.snackbarCopyImplicitAccountId') {
+            this.setState({
+                snackBarMessage: 'setupSeedPhrase.snackbarCopyImplicitAccountId'
+            })
+        }
+    }
+
+    componentDidMount = () => {
+        this.connection = nearApiJs.Connection.fromConfig({
+            networkId: NETWORK_ID,
+            provider: { type: 'JsonRpcProvider', args: { url: NODE_URL + '/' } },
+            signer: {},
+        })
+
+        const recoveryKeyPair = JSON.parse(localStorage.getItem(IMPLICIT_ACCOUNT_KEY) || '{}')
+
+        if (!recoveryKeyPair.publicKey) {
+            if (this.props.location.pathname === '/create-implicit') {
+                this.refreshData()
+            }
+        } else {
+            this.setState({ recoveryKeyPair })
+        }
+    }
+
+    refreshData = () => {
+        const { seedPhrase, publicKey, secretKey } = generateSeedPhrase()
+        const recoveryKeyPair = KeyPair.fromString(secretKey)
+        const wordId = Math.floor(Math.random() * 12)
+
+        this.setState((prevState) => ({
+            ...prevState,
+            seedPhrase,
+            publicKey,
+            wordId,
+            enterWord: '',
+            requestStatus: null,
+            recoveryKeyPair
+        }))
+    }
+
+    handleChangeWord = (e, { name, value }) => {
+        if (value.match(/[^a-zA-Z]/)) {
+            return false
+        }
+
+        this.setState((state) => ({
+           [name]: value.trim().toLowerCase(),
+           requestStatus: null
+        }))
+    }
+
+    handleStartOver = e => {
+        const {
+            history
+        } = this.props
+
+        this.refreshData()
+        history.push(`/create-implicit`)
+    }
+
+    handleSubmit = async () => {
+        const {
+            history
+        } = this.props
+        const { seedPhrase, enterWord, wordId, recoveryKeyPair } = this.state
+        if (enterWord !== seedPhrase.split(' ')[wordId]) {
+            this.setState(() => ({
+                requestStatus: {
+                    success: false,
+                    messageCode: 'account.verifySeedPhrase.error'
+                }
+            }))
+            return false
+        }
+        console.log('verified', recoveryKeyPair)
+        recoveryKeyPair.seedPhrase = seedPhrase
+        recoveryKeyPair.implicitAccountId = Buffer.from(recoveryKeyPair.publicKey.data).toString('hex')
```

Comment:
best not to save copies of values which can be derived from master value (or you risk weird bugs when they get out of sync). Like in this case saving private key is enough. Both public key and implicit account ID can be derived from private key. 

same with seed phrase – if you are saving seed phrase, saving anything else makes no sense – it can be computed from seed phrase.